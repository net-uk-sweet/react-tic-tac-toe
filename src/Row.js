import React from 'react';

export default function Row(props) {
    return (
        <div className="board-row">
            {props.children}
        </div>     
    )
  }