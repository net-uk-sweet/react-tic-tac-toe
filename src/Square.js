import React from 'react';

export default function Square(props) {
    const classes = props.highlight ? 'square highlight' : 'square';
    return (
      <button 
        className={classes}
        onClick={props.onClick}>
          {props.value}
      </button>
    );
  }