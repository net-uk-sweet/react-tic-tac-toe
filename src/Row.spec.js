import React from 'react';
import { mount, shallow } from 'enzyme';

import Row from './Row';
import Square from './Square';

let shared;

beforeEach(() => {
    shared = {
        children: [
            mount(<Square key='1' />),
            mount(<Square key='2' />),
            mount(<Square key='3' />)
        ]
    }
    shared.wrapper = shallow(<Row {...shared} />);
});

it('matches snapshot', () => {
    expect(shared.wrapper.getElement()).toMatchSnapshot();
});

it('renders it\s children', () => {
    expect(shared.wrapper.find('Square').length).toEqual(3);
});
