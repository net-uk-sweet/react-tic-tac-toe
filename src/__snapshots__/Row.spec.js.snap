// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="board-row"
>
  ReactWrapper {
    Symbol(enzyme.__unrendered__): <Square />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Object {
        "instance": <button
          class="square"
  />,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "square",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": Array [
          undefined,
        ],
        "type": "button",
      },
      "type": [Function],
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": Object {
          "instance": <button
            class="square"
  />,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "square",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "button",
        },
        "type": [Function],
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getSnapshotBeforeUpdate": true,
          },
        },
      },
      "attachTo": undefined,
      "hydrateIn": undefined,
    },
  }
  ReactWrapper {
    Symbol(enzyme.__unrendered__): <Square />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Object {
        "instance": <button
          class="square"
  />,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "square",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": Array [
          undefined,
        ],
        "type": "button",
      },
      "type": [Function],
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": Object {
          "instance": <button
            class="square"
  />,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "square",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "button",
        },
        "type": [Function],
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getSnapshotBeforeUpdate": true,
          },
        },
      },
      "attachTo": undefined,
      "hydrateIn": undefined,
    },
  }
  ReactWrapper {
    Symbol(enzyme.__unrendered__): <Square />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Object {
        "instance": <button
          class="square"
  />,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "square",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": Array [
          undefined,
        ],
        "type": "button",
      },
      "type": [Function],
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": Object {
          "instance": <button
            class="square"
  />,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "square",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "button",
        },
        "type": [Function],
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getSnapshotBeforeUpdate": true,
          },
        },
      },
      "attachTo": undefined,
      "hydrateIn": undefined,
    },
  }
</div>
`;
